api/controller.py

```diff
@@ -142,29 +142,42 @@ def __init__(self, _db, lanes=None, testing=False):
                 sys.exit()
         self._db = _db
         self.testing = testing
-        library = Library.instance(self._db)
-        self.library_id = library.id
-        if isinstance(lanes, LaneList):
-            lanes = lanes
-        else:
-            lanes = make_lanes(library, lanes)
-        self.top_level_lane = self.create_top_level_lane(lanes)
-
         self.auth = Authenticator(self._db)
-        self.setup_circulation(library)
         self.__external_search = None
+        
+        # Track the Lane configuration for each library by mapping its
+        # short name to the top-level lane.
+        self.top_level_lanes = {}
+
+        # Create a CirculationAPI for each library.
+        self.circulation_apis = {}
+
+        for library in _db.query(Library):
+            if isinstance(lanes, LaneList):
+                if not self.testing:
+                    raise CannotLoadConfiguration(
+                        "Cannot pass Lane objects into CirculationManager when not in testing mode."
+                    )
+                lanes = lanes
+            else:
+                lanes = make_lanes(library, lanes)
+            
+            self.top_level_lanes[library.id] = (
+                self.create_top_level_lane(
+                    library, lanes
+                )
+            )
+
+            self.circulation_apis[library.id] = self.setup_circulation(
+                library
+            )
+            self.setup_adobe_vendor_id(library)
         self.lending_policy = load_lending_policy(
             Configuration.policy('lending', {})
         )
 
         self.setup_controllers()
-        self.setup_adobe_vendor_id(library)
```

Comment:
It's simpler and faster to set up the Adobe Vendor ID configuration only when the test actually needs it.